corehq/apps/locations/util.py

```diff
@@ -123,18 +131,30 @@ def get_location_data_model(domain):
 
 class LocationExporter(object):
 
-    def __init__(self, domain, include_consumption=False):
+    def __init__(self, domain, include_consumption=False, async_task=None):
         self.domain = domain
         self.domain_obj = Domain.get_by_name(domain)
         self.include_consumption_flag = include_consumption
         self.data_model = get_location_data_model(domain)
         self.administrative_types = {}
+        self.location_types = self.domain_obj.location_types
+        self.async_task = async_task
+        self._location_count = None
+        self._locations_exported = 0
 
     @property
     @memoized
     def consumption_dict(self):
         return build_consumption_dict(self.domain)
 
+    def _increment_progress(self):
+        if self._location_count is None:
+            self._location_count = SQLLocation.active_objects.filter(domain=self.domain).count()
+            self._progress_update_chunksize = max(10, self._location_count // 100)
+        self._locations_exported += 1
+        if self._locations_exported % self._progress_update_chunksize == 0:
+            DownloadBase.set_progress(self.async_task, self._locations_exported, self._location_count)
```

Comment:
This wasn't implemented at all before, right?  I know some GS people who'll be very excited about that.

Reply:
Correct